### ISSUE TITLE
Having two order states with the same name makes status name disappear in order list after order validation

### ISSUE THREAD

**Issue by gennaris** [PrestaShop/PrestaShop#22870]
#### Describe the bug

Having two order states with the same name makes order status disappear in order list after order validation.

#### Steps to Reproduce

1 - Use the same name for two order statuses for the same id_lang 

(this is already happening in some existsings translations like IT where "On backorder (paid)" and "On backorder (unpaid)" are both translated in "In attesa di rifornimento". )

![1](https://user-images.githubusercontent.com/43071148/104894637-3f1f6580-5975-11eb-8079-6cb6d44c9586.png)

2 - Put a product to 0 quantity with "Allow orders when out of stock

3 - Make an order of that product and check order list afterwards :

![2](https://user-images.githubusercontent.com/43071148/104894647-434b8300-5975-11eb-8f6e-9c46efe5f147.png)

#### Additional information

* PrestaShop version: 1.7.7.1
* PHP version: 7.3

**Comment by hibatallahAouadni** [comment]
Hello @gennaris 

I didn't manage to reproduce your issue with PS1.7.7.1, see the attached screenshot below:

![image](https://user-images.githubusercontent.com/23308427/104918618-4fdfd380-5995-11eb-842e-47770d27dd37.png)

Please check and feedback.
Thanks!

**Comment by gennaris** [comment]
@hibatallahAouadni  I can't see your order list in the attached screenshot.

Anyway , in order to surely reproduce follow my instructions :  installing native Italian Language Pack and follow my instructions by using the "Out of stock" order state in italian language. That's how I discovered the issue.

**Comment by hibatallahAouadni** [comment]
Hello @gennaris 

Thanks for clarification :wink: 
I managed to reproduce the issue with the default language "English" and with PS1.7.7.0 & PS1.7.7.1, see the attached screenshot below:

![image](https://user-images.githubusercontent.com/23308427/104930889-fd5ae300-59a5-11eb-978d-bf2be1205197.png)

And it's a regression (I didn't reproduce it with PS1.7.6.9). I’ll add this to the debug backlog so that it’s fixed.
If you have already fixed it on your end or if you think you can do it, please do send us a pull request!

Thanks!

**Comment by hibatallahAouadni** [comment]
Ping @PululuK any idea about this issue's source?

**Comment by Progi1984** [comment]
The problem is only on the Italian language.

Why ? Because two strings in english have been translated by the same string.

:notebook: The problem is relative to Order Status (9 and 12).

**In english :gb:**
![image](https://user-images.githubusercontent.com/1533248/105373054-8d976300-5c06-11eb-8449-13d3537e8449.png)

**In italian :it:**
![image](https://user-images.githubusercontent.com/1533248/105373143-a43dba00-5c06-11eb-9352-5f0c0ed98bf4.png)

**Comment by Julievrz** [comment]
I corrected the translations on Crowdin.

See: 
https://crowdin.com/translate/prestashop-official/all/en-it?filter=basic&value=0#q=On%20back%20orders
https://crowdin.com/translate/prestashop-official/all/en-it?filter=basic&value=0#q=On%20back%20orders

It should be automatically updated in a few days.

**Comment by Progi1984** [comment]
@Julievrz Thanks. I will check on the next week ;)

**Comment by Julievrz** [comment]
> @Julievrz Thanks. I will check on the next week ;)

Sounds great, thank you!

**Comment by Progi1984** [comment]
Checked today in https://i18n.prestashop.com/translations/1.7.7.0/it-IT/it-IT.zip > `AdminOrderscustomersFeature.it-IT.xlf`

```
      <trans-unit id="48eac0bd985bc909fb58ccc1e1f669ed" approved="yes">
        <source>On backorder (paid)</source>
        <target state="final">In attesa di rifornimento</target>
        <note>Line: 42</note>
      </trans-unit>
      <trans-unit id="6d59fd4089482ca1a4017ca4852f8d1d" approved="yes">
        <source>On backorder (not paid)</source>
        <target state="final">In attesa di rifornimento</target>
        <note>Line: 44</note>
      </trans-unit>
```

Status : :heavy_multiplication_x:

**Comment by Julievrz** [comment]
![image](https://user-images.githubusercontent.com/70583503/105839328-9a82d080-5fd1-11eb-957e-f9f2f93a0126.png)
![image](https://user-images.githubusercontent.com/70583503/105839366-a7072900-5fd1-11eb-8952-a0894fbdbf51.png)

And yet I validated the strings 5 days ago... Isn't it related to this issue https://github.com/PrestaShop/PrestaShop/issues/22531 ?

**Comment by Progi1984** [comment]
@Julievrz Files are good here : https://github.com/PrestaShop/TranslationFiles/blob/master/1.7/translations/1.7.7/it-IT/AdminOrderscustomersFeature.it-IT.xlf#L765-L774 (Good news)

**Comment by Progi1984** [comment]
@PrestaShop/qa-functional Could you test this issue ? The `it-IT` language has been fixed.
@hibatallahAouadni

**Comment by hibatallahAouadni** [comment]
Hello @Progi1984 

I just tested with PS1.7.7.x and that's what I get:

1. BO>SELL>Order
2. Set the first order to **On backorder (not paid)**
3. Set the second order to **On backorder (paid)**
4. BO>CONFIGURE>Shop Parameters>Order Settings>Statuses
5. Change the status **On backorder (not paid)** to **On backorder (paid)**

![image](https://user-images.githubusercontent.com/23308427/106155968-533b4200-6181-11eb-850b-10563ceef134.png)

6. Then check BO>SELL>Order => See error

![image](https://user-images.githubusercontent.com/23308427/106156061-6221f480-6181-11eb-8ed2-de64c76e0f58.png)

7. Try to click on Order status select => See error: you will only find only one status

![image](https://user-images.githubusercontent.com/23308427/106156246-97c6dd80-6181-11eb-941f-ddb9efb0b720.png)

Please check and feedback.
Thanks!

**Comment by Julievrz** [comment]
> @Julievrz Files are good here : https://github.com/PrestaShop/TranslationFiles/blob/master/1.7/translations/1.7.7/it-IT/AdminOrderscustomersFeature.it-IT.xlf#L765-L774 (Good news)

Hi @Progi1984, It does not seem to work. The translation on the software is not updated.
![image](https://user-images.githubusercontent.com/70583503/106269398-bfbf4b00-622c-11eb-8023-4233cc552065.png)

**Comment by Progi1984** [comment]
@PrestaShop/product-team Is it logical that two order statuses have the same label ?

Because our choices providers have labels as keys.

**Comment by Julievrz** [comment]
> @PrestaShop/product-team Is it logical that two order statuses have the same label ?
> 
> Because our choices providers have labels as keys.

It's not exactly the same wording though? 🤔 Because in one case the backorder is already paid and in the other, it is not.

**Comment by MatShir** [comment]
If I understand well the bug:

We have the A and B orders with the status pizza and pasta as orders statuses
So we have the following order with their status:
A - 1 id pizza
B - 2 id pasta

We change the name of the pizza status to pasta.

So we have the following orders on the order list:
A - 1 id pizza (but it should have been 1 id pasta)
B - 2 id pasta 


So when we change the status of the order A, to the status order id 2 pasta. 
The display of the order status label is broken for the order A.
A - 2 id "nothing is displayed"
B - 2 id pasta 

Well, I don't understand why the label don't rely on the id. 
An order status' name can be changed, it might happen to create the order status with same label. 

The ideal is to be able to handle this case, with the management of the ID or not. It can be problematic since to remove this empty status, the status has to be changed again.

@Progi1984 What do you think we can do technically here for 1.7.7.2 ? 

(Sorry for the food example, it is lunchtime ^^)

**Comment by MatShir** [comment]
What have been discovered and discussed with @Progi1984 is that the selector on the order list can only display unique status name, so we are going to force the user on the edit/add form to enter a unique status name. 
The constraint will only be applied for the new status or fresh edited status.

**Comment by Julievrz** [comment]
Since we cannot add new wording in patch versions, I suggest we use the following existing string for now:

'This name already exists.' in Admin.Design.Notification 

The domain is not the right one and the error message could be better, so I suggest we improve the wording in 1.7.8.

**Comment by Progi1984** [comment]
Fixed by #23142
